### DISCUSSION TITLE
Webhook .json to GitLab overview

### DISCUSSION THREAD

**Discussion by lfkdev** [netbox-community/netbox#6635] [Q&A] [answered]
I currently use netboxs webhooks to automatically upload all changed data in netbox to our Git. 
However, the problem I'm currently facing is when e.g. something is changed in a device's interface, it is sometimes difficult to assign the resulting .json file to the webhook with the correct server in Git because the .json structure is always completely different depending on what you change. Are there any solutions to this or maybe already better existing workflows?

The goal is to have a clean auto-updating overview of all servers and their status (interfaces,name,dns etc) in GitLab.

**Comment by candlerb** [comment, marked as answer]
> the .json structure is always completely different depending on what you change

Can you give an example of what you're seeing, i.e. some JSON samples, and what you want to do with them?

It's pretty clear that whether you change a Device, an Interface, or an IP Address, you'll get different JSON, because those are different types of object.  What's the problem? Are you trying to associate those back to a Device?

Your webhook can make calls back to the Netbox API if required.  For example, given an IP address with an assigned interface_id, you can lookup that interface to map it to a device.  At the most basic level, you can just use the webhook as a trigger, and make API calls to fetch all the other information of interest (e.g. refetch the given device, all its interfaces, and all its IPs)

**Reply by lfkdev** [reply]
Yes, I need to associate those Json files back to the original device.
.json from a changed IP:

```json
{
   "event":"updated",
   "timestamp":"2021-06-21 12:06:57.889227+00:00",
   "model":"ipaddress",
   "username":"ooadmin",
   "request_id":"ea0a6171-10aa-40e7-85af-a401b146cee6",
   "data":{
      "id":449,
      "url":"/api/ipam/ip-addresses/449/",
      "family":{
         "value":4,
         "label":"IPv4"
      },
      "address":"172.28.70.210/24",
      "vrf":null,
      "tenant":{
         "id":1,
         "url":"/api/tenancy/tenants/1/",
         "name":"onOffice Gmbh",
         "slug":"onoffice-gmbh"
      },
      "status":{
         "value":"active",
         "label":"Active"
      },
      "role":null,
      "assigned_object_type":"dcim.interface",
      "assigned_object_id":2386,
      "assigned_object":{
         "id":2386,
         "url":"/api/dcim/interfaces/2386/",
         "device":{
            "id":147,
            "url":"/api/dcim/devices/147/",
            "name":"a01.onoffice.de",
            "display_name":"a01.onoffice.de"
         },
         "name":"IPMI",
         "cable":362,
         "connection_status":{
            "value":true,
            "label":"Connected"
         }
      },
      "nat_inside":null,
      "nat_outside":null,
      "dns_name":"a01.onoffice.de",
      "description":"ipmi_ip",
      "tags":[
         
      ],
      "custom_fields":{
         
      },
      "created":"2020-12-17",
      "last_updated":"2021-06-21T12:06:57.783613Z"
   }
}
```

So one way to realise this is to grab "data -> id:449" and then make an api call with this id to get the correct device name.. i'll try this, thanks! But there is no typical workflow already that you know of right? Wondering if we are the only one wanting to do this.

**Reply by candlerb** [reply]
It depends what you want to know.  The basic info on the device (its name and id) is already there:

```
    "assigned_object":{
         "id":2386,  // this the interface
         "url":"/api/dcim/interfaces/2386/",  // interface
         "device":{
            "id":147,
            "url":"/api/dcim/devices/147/",
            "name":"a01.onoffice.de",
            "display_name":"a01.onoffice.de"
         },
```

Notice the three levels: IP assigned to interface, interface belongs to device.

If you want to know more about the device, you'd need to make an api call with device id 147.
